cmd/singleplayer/main.go

```diff
@@ -1,33 +1,143 @@
 	return rand.Intn(2) == 1
 }
 
 func defectCooperateTwiceThenDefect(oa []bool, opa []bool) bool {
-	return len(oa) % 3 != 0
+	return len(oa)%3 != 0
+}
+
+func titForTat(oa []bool, opa []bool) bool {
+	n := len(oa)
+	if n < 1 {
+		return true
+	}
+	return opa[n-1]
+}
+
+func titForTatWithForgiveness(oa []bool, opa []bool) bool {
+	n := len(oa)
+	if n < 1 {
+		return true
+	}
+	// 10% chance to forgive a defection
+	if !opa[n-1] && rand.Float64() < 0.1 {
+		return true
+	}
+	return opa[n-1]
+}
+
+func megatron(oa []bool, opa []bool) bool {
+	if len(oa) < 10 {
+		return rand.Intn(2) == 1
+	}
+
+	for _, a := range opa {
+		if !a {
+			return false
+		}
+	}
+	return rand.Intn(2) == 1
 }
 
 func CreatePrisoners() []pd.Prisoner {
 	prisoners := []pd.Prisoner{
+		// Jackson
 		{Name: "Random", Owner: "Jackson", Strategy: random},
 		{Name: "Mod 3 Prisoner", Owner: "Jackson", Strategy: defectCooperateTwiceThenDefect},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+
+		// Hunter
+		{Name: "Tit for Tat", Owner: "Hunter", Strategy: titForTat},
+		{Name: "Megaton", Owner: "Hunter", Strategy: megatron},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+
+		// ChatGPT
+		{Name: "T4T Forgiveness", Owner: "ChatGPT", Strategy: titForTatWithForgiveness}, // Alex
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
+		{Name: "Defect", Owner: "Hunter", Strategy: func([]bool, []bool) bool { return false }},
 	}
 	return prisoners
 }
 
+func prettyPrintMap(m map[string]int, title string) {
+	// Extract keys and sort them
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	// Print the map in a pretty format
+	fmt.Printf("%s:\n", title)
+	fmt.Println("---------------")
+	for _, key := range keys {
+		fmt.Printf("%-15s : %d\n", key, m[key])
+	}
+}
+
+// Function to calculate and print average scores
+func printAverageScores(totals map[string]int, counts map[string]int, title string) {
+	keys := make([]string, 0, len(counts))
+	for key := range counts {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	fmt.Printf("%s:\n", title)
+	fmt.Println("---------------")
+	for _, key := range keys {
+		average := float64(totals[key]) / float64(counts[key])
+		fmt.Printf("%-15s : %.2f\n", key, average)
+	}
+}
```

Comment:
Can you make these methods you can call on []Game?

Reply:
Not without moving all of the logic in `main` into `game.go`

Reply:
Yeah id like it in game

Reply:
Having totals and ototals as a method off of []Game would be cleaner then we wouldn't need those maps